src/heitech.zer0mqXt.core/Main/Socket.cs

```diff
@@ -72,5 +76,32 @@ public void RegisterSubscriber<TMessage>(Action<TMessage> callback, Cancellation
             if (result.IsSuccess == false)
                 throw result.Exception;
         }
+
+        public async Task SendAsync<TMessage>(TMessage message) 
+            where TMessage : class, new()
+        {
+            var xtResult = await _sendReceive.SendAsync(message);
+
+            if (xtResult.IsSuccess == false)
+                throw xtResult.Exception;
```

Comment:
Did you intend to replace the `StackTrace` when throwing in this type? If we should throw with the stack trace information from the original Exception instead, consider using `ExceptionDispatchInfo.Capture(xtResult.Exception).Throw();`